src/ssh.c

```diff
@@ -187,14 +193,22 @@ int dcc_ssh_connect(char *ssh_cmd,
                     pid_t *ssh_pid)
 {
     pid_t ret;
-    char *child_argv[10];
-    int i;
+    int i,j;
+    char *ssh_args[6];
+    int num_ssh_args = 0;
 
     /* We need to cast away constness.  I promise the strings in the argv[]
      * will not be modified. */
 
-    if (!ssh_cmd)
-        ssh_cmd = getenv("DISTCC_SSH");
+    if (!ssh_cmd) {
+        char *ssh_cmd_in = getenv("DISTCC_SSH");
```

Comment:
Please could all of the variables be declared at the start of the function, as there were previously.